src/algorithms/signal_source/adapters/ion_gsms_signal_source.cc

```diff
@@ -0,0 +1,121 @@
+/*!
+ * \file ion_gsms_signal_source.h
+ * \brief GNSS-SDR Signal Source that reads sample streams following ION's GNSS-SDR metadata standard
+ * \author Víctor Castillo Agüero, 2024. victorcastilloaguero(at)gmail.com
+ *
+ * -----------------------------------------------------------------------------
+ *
+ * GNSS-SDR is a Global Navigation Satellite System software-defined receiver.
+ * This file is part of GNSS-SDR.
+ *
+ * Copyright (C) 2010-2020  (see AUTHORS file for a list of contributors)
+ * SPDX-License-Identifier: GPL-3.0-or-later
+ *
+ * -----------------------------------------------------------------------------
+ */
+
+#include "ion_gsms_signal_source.h"
+#include "gnss_sdr_flags.h"
+#include "gnss_sdr_string_literals.h"
+#include <gnuradio/blocks/copy.h>
+#include <string>
+#include <unordered_set>
+#include <vector>
+
+#if USE_GLOG_AND_GFLAGS
```

Comment:
This is the logging and flags mechanism that GNSS_SDR uses. Taken right from `file_source_base.cc`. But I may have been too excitedly adding it to header files too instead of just source files.

I saw @carlesfernandez took care of that in his branch. I will merge it into my branch and make sure it's only included where it's needed.